packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +61,106 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const isValidDate = (date: Date) => date instanceof Date && !isNaN(date.valueOf());
+
+const calculateTimezoneByValueOld = (timezone: string, date = new Date()) => {
+  const customTimezones = timeZoneDataUtils.getTimeZonesOld();
+  if (customTimezones.length === 0) {
+    return undefined;
+  }
+
+  const dateUtc = createUTCDate(date);
+  return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+};
+
+const calculateTimezoneByValueCore = (timeZone: string, date = new Date()): number | undefined => {
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    errors.log('W0009', timeZone);
+    return undefined;
+  }
```

Comment:
Take a look at the ```getStringOffset``` function comment

```suggestion
  const offset = getStringOffset(timeZone, date);

  if (offset === undefined) {
    errors.log('W0009', timeZone);
    return undefined;
  }
```